api-configuration/src/main/java/org/jvalue/ods/apiConfigurationService/endpoint/ApiConfigurationService.java

```diff
@@ -0,0 +1,143 @@
+package org.jvalue.ods.apiConfigurationService.endpoint;
+
+import org.jvalue.ods.apiConfigurationService.helper.GraphqlQueryFactory;
+import org.jvalue.ods.apiConfigurationService.helper.SqlFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.Arrays;
+
+import static org.jvalue.ods.apiConfigurationService.helper.GraphqlQueryFactory.createSQLGraphQuery;
+
+@RestController
+@RequestMapping("/")
+public class ApiConfigurationService implements ApiConfigurationServiceI {
+
+  private static final String HASURA_ADMIN_ENDPOINT = "http://172.17.0.1:7654/v1/query";
+  private static final String DEFAULT_SCHEMA = "storage";
+
+  @PostMapping("test")
+  public HttpStatus testThis() {
+    new SqlFactory().createViewQueryWithUnpackedFields("3", "s");
+    return HttpStatus.CONFLICT;
+  }
+
+  @Override
+  @PostMapping("/deleteAPIForTable")
+  public HttpStatus deleteAPIForTable(@RequestParam(name = "tableName", required = true) String tableName) {
+    try {
+      String schemaName = buildSchemaName(tableName);
+      untrackTable(schemaName);
+      return HttpStatus.OK;
+    } catch (Exception e) {
+      throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  @Override
+  @PostMapping("/createDefaultApiForTable")
+  public HttpStatus createDefaultApiForTable(@RequestParam(name = "tableName", required = true) String tableName) {
+    try {
+      String schemaName = buildSchemaName(tableName);
+      String sqlQuery = new SqlFactory().createViewQueryWithUnpackedFields(tableName, schemaName);
+      sendQuery(HASURA_ADMIN_ENDPOINT, createSQLGraphQuery(sqlQuery));
+      trackTable(schemaName);
+      return HttpStatus.OK;
+    } catch (Exception e) {
+      throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, e.getMessage());
+    }
+  }
+
+  @Override
+  @PostMapping("/removeDefaultAPI")
+  public HttpStatus removeDefaultAPI(@RequestParam(name = "tableName", required = true) String[] tableNames) {
```

Comment:
I initially designed the service to be called via rest endpoints. However, since rabbitmq has been introduced this approach might be obsolete by now.
Is the idea with rabbitmq to completly switch from rest to messaging or are there still different use cases?
Foe example i don't know how rabbitmq can deal with responses, e.g. whether the creation of the API has been successful or not.
Would i just put this response in a different channel/topic then?

Reply:
Ok, so we currently do still have REST and AMQP as separate concepts.

REST is currently majorly used for interacting with the client.
AMQP is currently used for services communicating with each other.

I feel like we should talk about when the behavior typically is triggered in the system. We are using domain events (like PipelineConfigCreated, etc.) which could be a trigger to creating / deleting APIs. Probably we will want to discuss this in a little meeting ;)